setup.py

```diff
@@ -8,55 +8,98 @@
 
 Simple check list for release from AllenNLP repo: https://github.com/allenai/allennlp/blob/master/setup.py
 
-To create the package for pypi.
+Steps to make a release:
 
 0. Prerequisites:
    - Dependencies:
-     - twine: "pip install twine"
+     - twine: `pip install twine`
    - Create an account in (and join the 'datasets' project):
      - PyPI: https://pypi.org/
      - Test PyPI: https://test.pypi.org/
-   - Don't break `transformers`
-     - Run the `transformers` CI using the `main` branch and make sure it's green.
-       In `transformers`, use "datasets @ git+https://github.com/huggingface/datasets@main#egg=datasets"
-       in both setup.py and src/transformers/dependency_versions_table.py and then run the CI
+   - Don't break `transformers`: run the `transformers` CI using the `main` branch and make sure it's green.
+     - In `transformers`, use `datasets @ git+https://github.com/huggingface/datasets@main#egg=datasets`
+       in both:
+       - setup.py and
+       - src/transformers/dependency_versions_table.py
+     - and then run the CI
 
-1. Change the version in:
+1. Create the release branch from main branch:
+     ```
+     git checkout main
+     git pull upstream main
+     git checkout -b release
```

Comment:
(nit) in case a branch with this name already exists
```suggestion
     git checkout -b release-VERSION
```

Reply:
I made it this way to make it simpler: as remote branches are deleted after the PR is merged.

Reply:
If a local branch with the same name exists, then previous command generated this error:
```
fatal: A branch named 'release' already exists.
```

Reply:
oh ok !

Reply:
I have changed the release branch name so that it triggers docs generation: 88c9c83

Context:
- #5251

Reply:
@lhoestq after my arguments above, do you still think it is important the name of the branches to be unique for each release even if they are deleted after the PR is merged?

I make this question in relation with the other branch: `dev-version`

Reply:
it's fine to not have it unique then

Reply:
If we have it unique we won't need to delete the branch on a local clone of the repo though - the branch exists temporarily on upstream, but locally it's not deleted when the PR is merged

Reply:
The unique naming for the dev version would be more complicated though: something like `vX.X.X+1.dev0-dev-version`? 

Maybe it is easier to add a step before to delete the local branch `dev-version`?

Reply:
Yea let's add a step to delete the local branch then

Reply:
OK. That step was already added: [cbd5874](https://github.com/huggingface/datasets/pull/5250/commits/cbd587492d97c6fa5cb1326133f989625292438b)